### PR TITLE
feat(platform): complete guided learning and harden streaming, ingestion, and deploys

### PR DESCRIPTION
**Pull request by WilliamAGH**
## Summary

Java learners can now open every guided lesson, receive stable streamed answers from the explicitly selected LLM provider, and navigate without obsolete lesson or chat callbacks changing the active view. Ingestion and model-output contracts now fail clearly on ambiguous input, while local, CI, and container verification share one reproducible toolchain and deployments prove the exact source revision they run.

### Learner experience

- **Complete guided curriculum**: Every table-of-contents slug resolves to packaged lesson Markdown, so guided navigation no longer reaches missing resources (`GuidedLearningService`, `src/main/resources/guided/lessons/`).
- **Lifecycle-safe streaming**: Unmounted chat requests and replaced lesson or guided-chat requests are aborted; ownership guards prevent stale text, citations, status, provider details, errors, and scroll indicators from mutating the active view (`ChatView`, `LearnView`).
- **Stable terminal errors**: Backpressure overflow and provider cooldown end the stream with the shared retryable error contract, and partial assistant responses are not committed to chat history (`SseSupport`, `ChatController`, `GuidedLearningController`).
- **Faithful Markdown and citations**: Fenced and indented code retain their list context, Unicode-blank enrichments disappear, Javadoc member anchors remain distinct, and PDF query/fragment links retain their destinations (`MarkdownNormalizer`, `InlineListParser`, `RetrievalService`, `frontend/src/lib/utils/url.ts`).

### Provider, model, and ingestion contracts

- **Configured-provider-only dispatch**: Streaming and completion reserve capacity immediately before dispatch, share cooldown admission, and never silently route a failed request through a different provider (`OpenAiProviderRoutingService`, `OpenAIStreamingService`).
- **Strict request and model output**: Chat requires meaningful Unicode-visible `latest` input; enrichment accepts one complete typed JSON object; reranking requires an exact document permutation (`ChatStreamRequest`, `EnrichmentService`, `RerankerService`).
- **Deterministic documentation ingestion**: Malformed UTF-8 fails before chunking or Qdrant writes, manifest-backed Java API sources share one classifier, and provenance changes invalidate stale ingestion markers (`FileOperationsService`, `JavaPackageExtractor`, `IngestionProvenanceDeriver`).
- **Canonical catalogs**: Documentation sources, Java API mirrors, and enrichment presentation project their manifests instead of maintaining duplicate inventories (`documentation-sources.manifest`, `java-api-documentation-sources.manifest`, `enrichment-kinds.manifest`).

### Configuration, verification, and deployment

- **Fail-fast configuration without tracked secrets**: Provider selection, matching credentials, reasoning effort, embedding settings, and reranker settings are validated through their canonical owners; Qdrant and embedding credentials come directly from process environment (`AppProperties`, `RequiredCredentialValidation`, `QdrantClientConfig`, `RemoteEmbedding`).
- **Canonical remote embedding deployment**: The non-secret Novita endpoint and provider-qualified model are owned by `application.properties`; credentials activate the provider without restoring removed environment aliases, runtime and ingestion preflight apply the same selection semantics, and deployment-shaped startup tests cover both remote and explicit OpenAI selection (`EmbeddingConfig`, `RemoteEmbedding`, `embedding_preflight.sh`, `EmbeddingConfigStartupTest`).
- **One reproducible Java distribution**: Local version managers, Gradle, GitHub Actions, and backend container stages use BellSoft Liberica 25.0.3+11 (`.tool-versions`, `build.gradle.kts`, `.github/workflows/build.yml`, `Dockerfile`).
- **Warning-free serialized verification**: Backend and frontend build output is serialized through a cross-platform kernel file lock; lint covers production and test code, expected negative-path logs are explicit, and Ubuntu CI exercises the lock wrapper (`BuildStateLock`, `Makefile`, `ExpectedLogEvents`).
- **Exact deployment identity**: Runtime info exposes sanitized deployment/build metadata, and Docker smoke requires both commit fields to equal the GitHub Actions SHA before the deploy is accepted (`SecurityConfig`, `JavaChatApplicationTests`, `.github/workflows/build.yml`).
- **Current operational documentation**: Setup, provider selection, reranking, deferred Qdrant initialization, frontend CI, ingestion, and Markdown ownership guidance now describe the shipped architecture rather than removed fallback paths.

## Breaking changes

- Runtime LLM fallback is removed. Set `LLM_PRIMARY_PROVIDER` to `github_models` or `openai`, provide that provider's matching credential, and restart to switch providers.
- `POST /api/chat/stream` accepts `latest` only; clients using the legacy `message` field must migrate.
- SSE consumers read provider identity from the dedicated `provider` event; removed fallback attempt fields are no longer emitted.
- Malformed UTF-8 local documents fail ingestion instead of being replacement-decoded.
- Development and container Java environments require BellSoft Liberica 25.0.3+11.

## Validation

- Local pre-push gate: backend build and all 530 tests passed; SpotBugs, PMD, ast-grep, frontend lint, type-aware lint, Svelte diagnostics, and production frontend build passed with zero warnings.
- Frontend unit suite: 15 files and 133 tests passed.
- Concurrent build-lock smoke: two commands completed serially and propagated child exit status.
- GitHub Actions at `a214437`: both the [push run](https://github.com/WilliamAGH/java-chat/actions/runs/29627932463) and [pull-request run](https://github.com/WilliamAGH/java-chat/actions/runs/29627933571) passed frontend, Java build, Ubuntu build-lock smoke, and exact-SHA Docker runtime identity checks.
- Coolify dev deployment `hpkpo3o35k7uj9wamqqizn2f`: `deployment.commit` and `build.commit` both equal `a21443783082f4beddd1f3539c5fe2ffa51fa2a9`; health, readiness, liveness, and dependency health remained HTTP 200/`UP` across two checks 30 seconds apart.
- Live dogfood at `dev.javachat.ai`: cited chat streaming, guided lesson content, lesson-scoped chat, desktop/mobile navigation, and all observed API requests passed with no browser console or page errors.

### CODE REVIEW

File: src/main/java/com/williamcallahan/javachat/web/ChatStreamRequest.java [hunk 1]
```diff
@@ -2,23 +2,23 @@
...
      */
-    public String userQuery() {
-        if (message != null && !message.isEmpty()) {
-            return message;
+    public ChatStreamRequest {
+        if (latest == null || latest.isBlank()) {
+            throw new IllegalArgumentException(LATEST_QUERY_REQUIRED_MESSAGE);
         }
-        return latest != null ? latest : "";
     }
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/main/java/com/williamcallahan/javachat/web/ChatStreamRequest.java:24]: Implemented. ChatStreamRequest now validates with UnicodeVisibleContent, including regression coverage for NBSP and FEFF effectively blank input. The full backend suite and pre-push verification pass at e4d7e40.

File: frontend/src/lib/components/ChatView.svelte [hunk 2]
```diff
@@ -37,9 +38,24 @@
     // Streaming state from composable (with 800ms status persistence)
     const streaming = createStreamingState({ statusClearDelayMs: 800 });
 
-    // Cleanup timers on unmount
+    function cancelInFlightChatStream(): void {
+        activeChatStreamController?.abort();
+        activeChatStreamController = null;
+    }
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply frontend/src/lib/components/ChatView.svelte:44]: Verified as non-actionable for the supported lifecycle. This cancellation helper is called during component teardown and before a new idle send; teardown immediately calls streaming.cleanup, and no mounted cancel action invokes it alone. Every callback is also guarded by AbortController ownership, so an aborted stream cannot update the view.

File: frontend/src/lib/components/ChatView.svelte [hunk 3]
```diff
@@ -145,6 +176,9 @@
         if (!message.trim() || streaming.isStreaming) return;
 
         const userQuery = message.trim();
+        cancelInFlightChatStream();
+        const chatStreamController = new AbortController();
+        activeChatStreamController = chatStreamController;
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply frontend/src/lib/components/ChatView.svelte:181]: Verified as intentional single-stream UX. ChatInput is disabled while streaming and handleSend rejects concurrent sends, so user supersession is not a supported action. Teardown cancellation remains necessary, and AbortController ownership guards every asynchronous callback against stale updates.

File: docs/api.md [hunk 4]
```diff
@@ -15,6 +15,7 @@ Request body:
 SSE event types (see `SseConstants`):
 
 - `status` → `{"message":"...","details":"..."}`
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply docs/api.md]: Implemented. docs/api.md now documents message, details, code, retryable, and stage for the status payload, including nullability. The backend and frontend use the shared typed contract.

File: docs/api.md [hunk 5]
```diff
@@ -15,6 +15,7 @@ Request body:
...
+- `provider` → `{"provider":"GitHub Models"}` (the configured provider selected for this request)
 - `text` → `{"text":"..."}`
 - `citation` → JSON array of citations
 - `error` → `{"message":"...","details":"..."}`
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply docs/api.md]: Implemented. docs/api.md now documents message, details, code, retryable, and stage for the error payload, including nullability. The backend and frontend use the shared typed contract.

File: src/main/java/com/williamcallahan/javachat/service/ConfiguredProviderTemporarilyUnavailableException.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+final class ConfiguredProviderTemporarilyUnavailableException extends IllegalStateException {
+    private final RateLimitService.ApiProvider provider;
+
+    /**
+     * Creates a retryable provider-admission failure for the configured provider.
+     *
+     * @param provider provider that temporarily rejected admission
+     */
+    ConfiguredProviderTemporarilyUnavailableException(RateLimitService.ApiProvider provider) {
+        super(messageFor(provider));
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/main/java/com/williamcallahan/javachat/service/ConfiguredProviderTemporarilyUnavailableException.java:20]: Implemented. ChatController maps ConfiguredProviderTemporarilyUnavailableException through SseSupport.configuredProviderUnavailableError to a stable retryable terminal SSE error. Guided learning uses the same mapping, and focused failure tests pass.

File: .config/lefthook.yml [hunk 7]
```diff
@@ -7,7 +7,7 @@ pre-commit:
       stage_fixed: true
     frontend-format:
       root: "frontend/"
-      glob: "*.{js,ts,svelte}"
+      glob: "*.{js,ts}"
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply .config/lefthook.yml:10]: Verified against installed oxfmt 0.27.0: it does not accept Svelte targets and exits with Expected at least one target file. The hook therefore limits oxfmt to supported JavaScript and TypeScript files; Svelte files are covered by svelte-check, type-aware lint, component tests, and the production build.

File: src/main/java/com/williamcallahan/javachat/service/ProviderCircuitState.java [hunk 8]
```diff
@@ -2,51 +2,67 @@
...
         return true;
     }
 
     /**
-     * Marks one successful request and closes the circuit if it was open.
+     * Closes the circuit after a successful request whose daily admission was already reserved.
      */
     synchronized void recordSuccess() {
         backoffMultiplier = 1;
         circuitOpen = false;
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/main/java/com/williamcallahan/javachat/service/ProviderCircuitState.java:65]: Verified as non-actionable in the complete admission design. A 429 records rateLimitedUntil in RateLimitState. recordSuccess does not clear that deadline, and RateLimitService checks persistent availability before consulting ProviderCircuitState, so a late success cannot reopen admission before the authoritative deadline.

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -31,10 +31,9 @@ spring.ai.openai.chat.options.temperature=${OPENAI_TEMPERATURE:0.7}
 # LLM-gateway priority class for live chat turns; untagged requests fall to the
 # gateway queue's zero-reservation "default" tier. Batch callers must override.
 spring.ai.openai.chat.options.http-headers.[X-Tier]=${LLM_GATEWAY_TIER:production-z}
-# Use GitHub token for GitHub Models, or OpenAI API key for fallback
-# To avoid rate limits, set OPENAI_API_KEY in addition to GITHUB_TOKEN
-spring.ai.openai.api-key=${GITHUB_TOKEN:${OPENAI_API_KEY:}}
-spring.ai.openai.chat.api-key=${GITHUB_TOKEN:${OPENAI_API_KEY:}}
+# This compatibility client uses GitHub Models only; it never borrows an OpenAI key.
+spring.ai.openai.api-key=${GITHUB_TOKEN:}
+spring.ai.openai.chat.api-key=${GITHUB_TOKEN:}
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/main/resources/application.properties]: Implemented. Tracked Spring AI OpenAI credential and base URL aliases were removed. The selected SDK clients now bind GitHub Models and OpenAI credentials and endpoints directly from their provider-specific process environment settings, with startup validation.

File: src/test/java/com/williamcallahan/javachat/service/EnrichmentServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+    private OpenAIStreamingService openAIStreamingService;
+    private EnrichmentService enrichmentService;
+
+    @BeforeEach
+    void setUp() {
+        openAIStreamingService = mock(OpenAIStreamingService.class);
+        when(openAIStreamingService.isAvailable()).thenReturn(true);
+        enrichmentService = new EnrichmentService(new ObjectMapper(), openAIStreamingService);
+    }
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/test/java/com/williamcallahan/javachat/service/EnrichmentServiceTest.java:49]: Implemented. EnrichmentServiceTest now obtains its ObjectMapper from focused Spring configuration instead of constructing a bare mapper, preserving production module and coercion behavior.

File: src/test/java/com/williamcallahan/javachat/service/OpenAiRequestFactoryTest.java [hunk 11]
```diff
@@ -1,17 +1,93 @@
...
+        application.setLazyInitialization(true);
+        application.setLogStartupInfo(false);
+        application.setRegisterShutdownHook(false);
+
+        RuntimeException startupFailure =
+                assertThrows(RuntimeException.class, () -> application.run("--OPENAI_REASONING_EFFORT=hgh"));
+
+        IllegalArgumentException configurationFailure = findConfigurationFailure(startupFailure);
+        assertTrue(configurationFailure.getMessage().contains("Invalid OPENAI_REASONING_EFFORT value 'hgh'"));
+    }
```
[the remark this answers is not in the document]
Comment (WilliamAGH) [reply src/test/java/com/williamcallahan/javachat/service/OpenAiRequestFactoryTest.java:80]: Implemented. Reasoning effort is owned by the canonical app.llm.reasoning-effort Spring property in AppProperties and application.properties. The startup regression test now uses that property rather than an environment-style alias.
